### PR TITLE
Normalize URLs containing www subdomain

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes issue where the output of the `site_url` and `home_url` filters could change depending on the context in which a sync was initialised.

In particular, sites might alternate between www.example.com and example.com.

This change _only_ targets the specific case of the www subdomain, as I think it's pretty universal that it's the same site as the naked domain. A case where the URL switches to, say, blog.example.com is harder to make assumptions about. Did they just permanently reconfigure their web server, or is this a temporary change specific to this request?

### CODE REVIEW

File: tests/php/sync/test_class.jetpack-sync-callables.php [hunk 1]
```diff
@@ -164,5 +164,38 @@ function test_preserve_scheme() {
 		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://site.com' ), 'https://site.com' );
 		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://site.com/blog' ), 'https://site.com/blog' );
 		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://example.org' ), 'https://example.org' );
+
+		// adding www subdomain reverts to original domain
+		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://www.example.com', true ), 'https://example.com' );
```
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-callables.php]: I think if the url in the option is set to be` http://`
we should convert `https://www.example.com` to `http://example.com`

**Comment by enejb** [comment]
This makes sense to me. 👍 
Although it might also make sense to have the resverse of that. 
http://www.example.com in the option shouldn't be switched to http://example.com 
Since whatever is in the option should be considered the preferred url.

File: tests/php/sync/test_class.jetpack-sync-callables.php (hunk 1 above)
Comment (gravityrail) [inline tests/php/sync/test_class.jetpack-sync-callables.php]: Yes, but in this case a few lines up the option got switched to https

**Comment by gravityrail** [comment]
@enejb - great idea. I just pushed that change.

File: tests/php/sync/test_class.jetpack-sync-callables.php (hunk 1 above)
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-callables.php]: ups I didn't see that :)

**Pull request merged**